python/GafferUI/StringPlugValueWidget.py

```diff
@@ -91,19 +93,34 @@ def _updateFromPlugs( self ) :
 
 		self.__textWidget.setErrored( errored )
 
-		self.__textChangedConnection.block(
-			any( not Gaffer.Metadata.value( p, "stringPlugValueWidget:continuousUpdate" ) for p in self.getPlugs() )
-		)
+		self.__continuousUpdate = all( Gaffer.Metadata.value( p, "stringPlugValueWidget:continuousUpdate" ) for p in self.getPlugs() )
 
 		placeHolder = ""
 		if value is None and len( self.getPlugs() ) :
 			placeHolder = "---"
+			# Mixed values require interaction before we commit the widget
+			# value to the plugs. This prevents mixed values being overriden
+			# by the empty string in the widget when it loses focus.
+			self.__editRequiresInteraction = True
 		else :
-			placeHolder = sole( Gaffer.Metadata.value( p, "stringPlugValueWidget:placeholderText" ) for p in self.getPlugs() ) or ""
+			placeHolder = self.__placeholderText()
+			self.__editRequiresInteraction = False
 		self.textWidget()._qtWidget().setPlaceholderText( placeHolder )
 
 		self.__textWidget.setEditable( self._editable() )
 
+	def __editInteractionOccured( self ) :
```

Comment:
Do you think there's merit in moving this logic to TextWidget, so that we don't even emit `editingFinished()` if there has been no interaction? The docs for `TextWidget.editingFinishedSignal()` do seem to give that impression.

Reply:
Does it depend slightly on whether you include 'focusing the edit field' as part of 'edited'? Eg: gave the widget focus with intent by clicking in it, then changed their mind.  Happy to move if you like, or save it for later when we can mull some more?

Reply:
Yeah, let's save for later.